### PR TITLE
Fix issue 247: detect brs in text to create readerable content

### PR DESCRIPTION
**Pull request by gijsk**
@leibovic How does this look for #247 ? I've not really done a serious performance test, and the messing with the array is a bit icky... but otherwise I think this should do? I picked 5 somewhat arbitrary (see the issue IRC log), but that seems like an OK start to me, and we can tweak if necessary...

### CODE REVIEW

File: Readability.js [hunk 1]
```diff
@@ -1703,6 +1703,17 @@ Readability.prototype = {
    */
   isProbablyReaderable: function(helperIsVisible) {
     var nodes = this._getAllNodesWithTag(this._doc, ["p", "pre"]);
+    if (this._doc.querySelectorAll && Array.from) {
+      nodes = Array.from(nodes);
+      var brs = this._doc.querySelectorAll("br:nth-of-type(5)");
```
Comment (leibovic) [inline Readability.js]: I would at least leave a comment explaining that 5 is a bit of an arbitrary value here.

File: Readability.js [hunk 2]
```diff
@@ -1703,6 +1703,17 @@ Readability.prototype = {
...
+      for (var i = 0; i < brs.length; i++) {
```
Comment (leibovic) [inline Readability.js]: Should we limit this to more than just brs.length? What if we have a page that has a ton of brs?

**Comment by leibovic** [comment]
This approach seems sane, but I do worry a bit about performance. Do we have any performance tests in place? I know we have telemetry histogram probes for parse times, but I wonder if it would be worth landing one for the `isProbablyReaderable` time as well.

**Comment by gijsk** [comment]
Fair points. The code is simpler this way, but it would probably be more performant if we avoided getting the nodelist length (as well as converting the one for the paragraphs into an array). I can look at that.

We do have performance tests for the commandline stuff but that won't work with this selector, I'm pretty sure, so I'll have to do some manual testing to get a better idea of what the impact is.

**Comment by salty-horse** [comment]
Bug #247 appears fixed in Firefox 52.0.1 - should this be closed?

**Comment by gijsk** [comment]
Yeah, closing makes sense at this point. Thanks!

**Pull request closed without merging**